src/Transport/HttpTransport.php

```diff
@@ -148,13 +146,19 @@ public function close(?int $timeout = null): PromiseInterface
      */
     private function cleanupPendingRequests(): void
     {
-        while ($promise = array_pop($this->pendingRequests)) {
-            try {
-                $promise->wait();
-            } catch (\Throwable $exception) {
-                // Do nothing because we don't want to break applications while
-                // trying to send events
-            }
+        try {
+            $requestGenerator = function () {
+                foreach ($this->pendingRequests as $key => $request) {
+                    yield $key => $this->httpClient->sendAsyncRequest($request);
+                }
+            };
+            $eachPromise = new EachPromise($requestGenerator(), ['concurrency' => 30]);
```

Comment:
Please add a blank line above this one to improve readability

Reply:
Done